### PR TITLE
[reference-device] send fake ADDR_NTF.ntf

### PR DESCRIPTION
**Pull request by simonlingoogle**
This PR allows Reference Devices to send fake Proactive Address Notification messages. 
This PR will be used by #5516 for testing purpose.

### CODE REVIEW

File: include/openthread/thread.h [hunk 1]
```diff
@@ -836,6 +836,23 @@ typedef void (*otThreadDiscoveryRequestCallback)(const otThreadDiscoveryRequestI
 void otThreadSetDiscoveryRequestCallback(otInstance *                     aInstnace,
                                          otThreadDiscoveryRequestCallback aCallback,
                                          void *                           aContext);
+
+/**
+ * This method sends a Proactive Address Notification (ADDR_NTF.ntf) message.
```
Comment (abtink) [inline include/openthread/thread.h:841]:
```suggestion
 * This function sends a Proactive Address Notification (ADDR_NTF.ntf) message.
```

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/core/thread/address_resolver.hpp [hunk 2]
```diff
@@ -177,6 +177,21 @@ class AddressResolver : public InstanceLocator
      */
     void RestartAddressQueries(void);
 
+    /**
+     * This method sends a Address Notification (ADDR_NTF.ans) message.
```
Comment (librasungirl) [inline src/core/thread/address_resolver.hpp:181]:
```suggestion
     * This method sends an Address Notification (ADDR_NTF.ans) message.
```

File: src/cli/README.md [hunk 3]
```diff
@@ -807,6 +808,19 @@ Delete all stored settings, and signal a platform reset.
 > factoryreset
 ```
 
+### fake
+
+Send fake Thread messages.
+
+Note: Only for certification test.
+
+#### fake /a/an \<dst-ipaddr\> \<target\> \<meshLocalIid\>
```
Comment (librasungirl) [inline src/cli/README.md:817]: It seems a bit weird to me as the name `meshLocalIid` seems representing an interface identifier, however below `::1111:2222:3333:4444` is an IPv6 address format.

Comment (simonlingoogle) [reply src/cli/README.md:817]: Yes, it's a bit wired but currently there is no method for parsing an IID.

I will use the similar solution of `bbr mgmt dua` to parse hex string (e.g. 1111222233334444) to IID using `Hex2Bin`.

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**